cardano_node_tests/tests/test_native_tokens.py

```diff
@@ -1873,6 +1875,128 @@ def test_transfer_no_ada(
                 )
         assert expected_error in str(excinfo.value)
 
+    @allure.link(helpers.get_vcs_link())
+    @hypothesis.given(
+        token_amount=st.integers(min_value=20_000_001, max_value=MAX_TOKEN_AMOUNT - 1)
+    )
+    @common.hypothesis_settings()
+    @pytest.mark.parametrize(
+        "use_build_cmd",
+        (
+            False,
+            pytest.param(
+                True,
+                marks=pytest.mark.skipif(not common.BUILD_USABLE, reason=common.BUILD_SKIP_MSG),
+            ),
+        ),
+        ids=("build_raw", "build"),
+    )
+    @pytest.mark.dbsync
+    def test_transfer_invalid_token_amount(
+        self,
+        cluster: clusterlib.ClusterLib,
+        payment_addrs: List[clusterlib.AddressRecord],
+        new_token: clusterlib_utils.TokenRecord,
+        use_build_cmd: bool,
+        token_amount: int,
+    ):
+        """Test sending an invalid amount of tokens to payment address."""
+        temp_template = f"{common.get_test_id(cluster)}_{token_amount}_{use_build_cmd}"
+
+        src_address = new_token.token_mint_addr.address
+        dst_address = payment_addrs[2].address
+
+        ma_destinations = [
+            clusterlib.TxOut(address=dst_address, amount=token_amount, coin=new_token.token),
+        ]
+
+        min_amount_lovelace = 1_379_280
+
+        destinations = [
+            *ma_destinations,
+            clusterlib.TxOut(address=dst_address, amount=min_amount_lovelace),
+        ]
+
+        tx_files = clusterlib.TxFiles(signing_key_files=[new_token.token_mint_addr.skey_file])
+
+        with pytest.raises(clusterlib.CLIError) as excinfo:
+            if use_build_cmd:
+                destinations.append(
+                    clusterlib.TxOut(address=src_address, amount=min_amount_lovelace)
+                )
+                cluster.build_tx(
+                    src_address=src_address,
+                    tx_name=temp_template,
+                    txouts=destinations,
+                    fee_buffer=2_000_000,
+                    tx_files=tx_files,
+                )
+            else:
+                cluster.send_funds(
+                    src_address=src_address,
+                    destinations=destinations,
+                    tx_name=temp_template,
+                    tx_files=tx_files,
+                )
+
+        if use_build_cmd:
+            assert "Non-Ada assets are unbalanced" in str(excinfo.value)
+        else:
+            assert "ValueNotConservedUTxO" in str(excinfo.value)
+
+    @pytest.mark.dbsync
+    def test_transfer_invalid_token_amount_inside_range(
+        self,
+        cluster: clusterlib.ClusterLib,
+        payment_addrs: List[clusterlib.AddressRecord],
+        new_token: clusterlib_utils.TokenRecord,
+        use_build_cmd: bool,
+        token_amount: int,
```

Comment:
Looks like you are missing hypothesis strategy for `token_amount`. And the `use_build_cmd` parametrization as well.